test/e2e/tests/dependencies.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/features"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/network"
+	mgmtnetwork "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-07-01/network"
+	mgmtfeatures "github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-07-01/features"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+)
+
+// EnvConfig contains all environment variables in one place
+type EnvConfig struct {
+	AzureSubscriptionID string
+	AzureTenantID       string
+	ResourceGroups      string
+	AROResourceGroup    string
+	Cluster             string
+	Location            string
+}
+
+// DepClientSet is used to control cluster dependencies like:
+// - resource groups
+// - network
+// - active directory
+type DepClientSet struct {
+	rgClientSet   features.ResourceGroupsClient
+	vnetClientSet network.VirtualNetworksClient
+	snetClientSet network.SubnetsClient
+	rtClientSet   network.RouteTablesClient
+}
+
+func readEnvConf() EnvConfig {
+	return EnvConfig{
+		AzureSubscriptionID: os.Getenv("AZURE_SUBSCRIPTION_ID"),
+		AzureTenantID:       os.Getenv("AZURE_TENANT_ID"),
+		ResourceGroups:      os.Getenv("RESOURCEGROUP"),
+		AROResourceGroup:    os.Getenv("ARO_RESOURCEGROUP"),
+		Cluster:             os.Getenv("CLUSTER"),
+		Location:            os.Getenv("LOCATION"),
+	}
+}
+
+// createResourceGroup creates test cluster resource group
+func createResourceGroup(ctx context.Context, dcs *DepClientSet, conf EnvConfig) error {
```

Comment:
One way to simplify all these create for different resources would be to use ARM (same as we do in deploy or cluster install) and use ARM client only.

Reply:
For thinking...